browser/ui/views/brave_help_bubble/brave_help_bubble_host_view.cc

```diff
@@ -0,0 +1,226 @@
+                              weak_factory_.GetWeakPtr()));
+}
+
+void BraveHelpBubbleHostView::UpdatePosition() {
+  auto browser_root_view_origin =
+      GetWidget()->GetRootView()->GetBoundsInScreen().origin();
+  auto tracked_element_origin =
+      tracked_element_->GetBoundsInScreen().CenterPoint();
+
+  auto circle_center = gfx::Point(kWidth / 2, kHeight / 2);
+
+  // Calculate the final origin point by taking into account the Browser
+  // window's position and frame's top padding
+  tracked_element_origin.set_x(tracked_element_origin.x() -
+                               browser_root_view_origin.x() -
+                               circle_center.x() + kBraveActionLeftMarginExtra);
+  tracked_element_origin.set_y(tracked_element_origin.y() -
+                               browser_root_view_origin.y() -
+                               circle_center.y());
+
+  SetPosition({tracked_element_origin.x(), tracked_element_origin.y()});
+}
+
+void BraveHelpBubbleHostView::StartObservingAndShow() {
+  // Observing changes to the parent element's bounds,
+  // as they are the only ones that are expected to change
+  if (!view_observation_.IsObserving()) {
+    view_observation_.Observe(tracked_element_->parent());
+  }
+
+  UpdatePosition();
+  Show();
+}
+
+void BraveHelpBubbleHostView::AddedToWidget() {
+  InitElementTrackers();
+
+  if (tracked_element_ && tracked_element_->GetVisible()) {
+    StartObservingAndShow();
+  }
+}
+
+void BraveHelpBubbleHostView::OnPaint(gfx::Canvas* canvas) {
+  cc::PaintFlags flags;
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setShader(kBraveGradient);
+  flags.setStrokeWidth(2.f);
+  canvas->DrawCircle(GetContentsBounds().CenterPoint(), 27.f, flags);
+  flags.setStrokeWidth(5.f);
+  canvas->DrawCircle(GetContentsBounds().CenterPoint(), 20.f, flags);
+}
+
+void BraveHelpBubbleHostView::OnBubbleDestroying(views::Widget* widget) {
+  // In the destruction, we don't have to remove this from parent. Otherwise,
+  // DCHECK will fail as the ancestor view is iterating its children to destroy
+  // all descendant.
+  if (GetWidget()->IsClosed()) {
+    return;
+  }
+
+  // There's a possibility that someone is iterating over the parent's children,
+  // and we are removing the child view early. This can cause a DCHECK in
+  // view.cc, so we schedule removal of the child view via TaskRunner to avoid
+  // this issue.
+  base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
+      FROM_HERE, base::BindOnce(
+                     [](base::WeakPtr<views::View> view) {
+                       if (!view) {
+                         return;
+                       }
+                       DCHECK(view->parent());
+                       view->parent()->RemoveChildViewT(view.get());
+                     },
+                     weak_factory_.GetWeakPtr()));
+
+  brave_help_bubble_delegate_view_ = nullptr;
```

Comment:
@petemill I have experienced two crashes with obscured stack traces, but I am unable to reproduce them. Both crashes occurred when the bubble closed. The goal here is to remove the View from the tree so it doesn't impact the memory during runtime. However, the fact that this sporadically crashes is much more riskier than the potential memory impact.

Perhaps we should simply turn off the visibility? Let me know if you have a better approach.

Reply:
What I'm wondering is why this would need to be handled any differently than other Views bubbles, e.g. Extension, MediaControls, or find-in-page? Can we just copy what they do, or see if there are any differences?